tests/test_decorator.py

```diff
@@ -80,127 +82,148 @@ def noauth_anonymous_view(request: HttpRequest) -> JsonResponse:
-                "/33/test?num=3&date=2020-01-01&string=abc&boolean=false&opt_num=1"
-                "&opt_date=2022-01-01"
-            ).status_code
-            == 200
-        )
-        # Include all optional values
-        assert (
-            client.get(
-                "/33/test?num=3&date=2020-01-01&string=abc&boolean=false&opt_num=1"
-                "&opt_date=2022-01-01"
-                "&opt_string=123&opt_boolean=1"
-            ).status_code
-            == 200
-        )
-        # Include only optional values but no required
-        assert (
-            client.get(
-                "/33/test?opt_num=1&opt_date=2022-01-01&opt_string=123&opt_boolean=1"
-            ).status_code
-            == 400
-        )
-        # Invalid path
-        assert (
-            client.get(
-                "/a33/test?num=3&date=2020-01-01&string=abc&boolean=false"
-            ).status_code
-            == 404
-        )
-        # Invalid number
-        assert (
-            client.get(
-                "/33/test?num=3x&date=2020-01-01&string=abc&boolean=false"
-            ).status_code
-            == 400
-        )
-        # Invalid date
-        assert (
-            client.get(
-                "/33/test?num=3&date=2020-31-31&string=abc&boolean=false"
-            ).status_code
-            == 400
-        )
-        # Invalid bool
-        assert (
-            client.get("/33/test?num=3&date=2020-01-01&string=&boolean=xyz").status_code
-            == 400
-        )
+    settings.ROOT_URLCONF = __name__
+    mocker.patch(f"{__name__}.urlpatterns", urls)
+    got = client.get(have_url)
+    assert got.status_code == want_status
+    if want_values is None:
+        collector.assert_not_called()
+    else:
+        collector.assert_called_once_with(mocker.ANY, **want_values)
+
+
+@pytest.mark.parametrize(
+    "view,path_spec,have_url,want_status,want_values",
+    [
+        (TestViews.path_string, "<str:pp>", "/hello", 200, {"pp": "hello"}),
+        (TestViews.path_string, "<str:pp>", "/", 404, None),
+        (TestViews.path_int, "<int:pp>", "/1", 200, {"pp": 1}),
+        (TestViews.path_int, "<int:pp>", "/h", 404, None),
+        (
+            TestViews.path_many,
+            "<str:pp1>/<int:pp2>",
+            "/hello/33",
+            200,
+            {"pp1": "hello", "pp2": 33},
+        ),
+    ],
+)
+def test_path_params(  # type: ignore
+    mocker, client, settings, view, path_spec, have_url, want_status, want_values
+):
+    """
+    Tests URL paths with the decorator. The decorator doesn't touch the parameters
+    currently, so we are essentially testing Django internals, but it is good to have
+    this check in case we start altering params.
+    """
+
+    collector, api_view = _create_api_view(view, None)  # type: ignore
+    urls = [
+        path(path_spec, api_view),
+    ]
+    settings.ROOT_URLCONF = __name__
+    mocker.patch(f"{__name__}.urlpatterns", urls)
+
+    got = client.get(have_url)
```

Comment:
```suggestion
    mocker.patch(f"{__name__}.urlpatterns", urls)

    with override_settings(ROOL_URLCONF=__name__):
        got = client.get(have_url)
```

Maybe something like this would be better? Otherwise you permanently modify the settings file, which could lead to confusing errors in other tests

Reply:
The settings fixture do revert any changes between tests.
https://github.com/pytest-dev/pytest-django/blob/d9b93568fb27b45340fa525c33112383ada66a3d/pytest_django/fixtures.py#L502

Reply:
Ah, sorry, I missed that this wasn't `django.conf.settings` 👍